include/picongpu/plugins/openPMD/openPMDWriter.hpp

```diff
@@ -0,0 +1,1417 @@
+
+            /* load all fields */
+            meta::ForEach< FileCheckpointFields, LoadFields< bmpl::_1 > >
+                ForEachLoadFields;
+            ForEachLoadFields( &mThreadParams );
+
+            /* load all particles */
+            meta::ForEach< FileCheckpointParticles, LoadSpecies< bmpl::_1 > >
+                ForEachLoadSpecies;
+            ForEachLoadSpecies( &mThreadParams, restartChunkSize );
+
+            IdProvider< simDim >::State idProvState;
+            ReadNDScalars< uint64_t, uint64_t >()(
+                mThreadParams,
+                "picongpu",
+                "idProvider",
+                "startId",
+                &idProvState.startId,
+                "maxNumProc",
+                &idProvState.maxNumProc );
+            ReadNDScalars< uint64_t >()(
+                mThreadParams,
+                "picongpu",
+                "idProvider",
+                "nextId",
+                &idProvState.nextId );
+            log< picLog::INPUT_OUTPUT >(
+                "Setting next free id on current rank: %1%" ) %
+                idProvState.nextId;
+            IdProvider< simDim >::setState( idProvState );
+
+            // avoid deadlock between not finished pmacc tasks and mpi calls in
+            // openPMD
+            __getTransactionEvent().waitForFinished();
+
+            // Finalize the openPMD Series by calling its destructor
+            mThreadParams.closeSeries();
+        }
+
+    private:
+        void
+        endWrite()
+        {
+            mThreadParams.fieldBuffer.resize( 0 );
+        }
+
+        void
+        initWrite()
+        {
+            // may be zero
+            auto size =
+                mThreadParams.window.localDimensions.size.productOfComponents();
+            mThreadParams.fieldBuffer.resize( size );
+        }
+
+        /**
+         * Notification for dump or checkpoint received
+         *
+         * @param currentStep current simulation step
+         */
+        void
+        dumpData( uint32_t currentStep )
+        {
+            // local offset + extent
+            const pmacc::Selection< simDim > & localDomain =
+                Environment< simDim >::get().SubGrid().getLocalDomain();
+            mThreadParams.cellDescription = m_cellDescription;
+            mThreadParams.currentStep = currentStep;
+
+            for( uint32_t i = 0; i < simDim; ++i )
+            {
+                mThreadParams.localWindowToDomainOffset[ i ] = 0;
+                if( mThreadParams.window.globalDimensions.offset[ i ] >
+                    localDomain.offset[ i ] )
+                {
+                    mThreadParams.localWindowToDomainOffset[ i ] =
+                        mThreadParams.window.globalDimensions.offset[ i ] -
+                        localDomain.offset[ i ];
+                }
+            }
+
+            /* copy species only one time per timestep to the host */
+            if( lastSpeciesSyncStep != currentStep )
```

Comment:
Please trigger this only if we use the double buffer strategy for dumping. 
Could you please extent the condition so that we only update if strategy==adios/double_buffering